File: CodeWars/JS/6kyu_real_password_cracker.js
// https://www.codewars.com/kata/59146f7b4670ba520900000a/train/javascript
const crypto = require("crypto");
function passwordCracker(hash) {
    const alphabet = " abcdefghijklmnopqrstuvwxyz";
    for (a of alphabet) {
        for (b of alphabet) {
            for (c of alphabet) {
                for (d of alphabet) {
                    for (e of alphabet) {
                        if (
                            crypto
                                .createHash("sha1")
                                .update(`${a}${b}${c}${d}${e}`.trim())
                                .digest("hex") === hash
                        ) {
                            return `${a}${b}${c}${d}${e}`.trim();
                        }
                    }
                }
            }
        }
    }
}

console.log(
    passwordCracker("e6fb06210fafc02fd7479ddbed2d042cc3a5155e") === "code",
);
console.log(
    passwordCracker("a94a8fe5ccb19ba61c4c0873d391e987982fbbd3") === "test",
);

File: CodeWars/python/6kyu_real_password_cracker.py
# https://www.codewars.com/kata/59146f7b4670ba520900000a/train/python
import codewars_test as test

"""Solution"""
from hashlib import sha1
from itertools import product


def password_cracker(hash):
    letters = ['', *map(lambda x:chr(x), range(97, 123))]
    for i in product(letters, repeat=5):
        if(sha1(''.join(i).encode()).hexdigest() == hash):
            return ''.join(i)


test.assert_equals(password_cracker(
    'e6fb06210fafc02fd7479ddbed2d042cc3a5155e'), 'code')
test.assert_equals(password_cracker(
    'a94a8fe5ccb19ba61c4c0873d391e987982fbbd3'), 'test')
